File: javascript/test/instance/GenericCollectionHolder_SizeAlias.ts
/*******************************************************************************
 Copyright (c) 2023-2024. Jonathan Bédard ~ JóôòKiwi

 This project is free to use.
 All the right is reserved to the author of this project.
 ******************************************************************************/

import type {StraightCollectionHolderForTest} from "./StraightCollectionHolderForTest"

import {GenericCollectionHolder} from "../../src/GenericCollectionHolder"
import {ABCD}                    from "../value/arrays"

export class GenericCollectionHolder_SizeAlias
    extends GenericCollectionHolder<string, readonly string[]>
    implements StraightCollectionHolderForTest<string> {

    public amountOfCall = 0

    public constructor() { super(ABCD,) }

    public execute(action: (instance: this,) => void,): this {
        action(this,)
        return this
    }

    public override get size(): number {
        this.amountOfCall++
        return super.size
    }

}

File: java/collection.joookiwi/src/test/java/instance/GenericCollectionHolder_SizeAlias.java
package instance;

import java.util.function.Consumer;
import joookiwi.collection.java.GenericCollectionHolder;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNullByDefault;
import org.jetbrains.annotations.Range;

import static joookiwi.collection.java.CommonContracts.ALWAYS_THIS_1;
import static joookiwi.collection.java.NumericConstants.MAX_INT_VALUE;
import static value.Arrays.ABCD;

@NotNullByDefault
public final class GenericCollectionHolder_SizeAlias
        extends GenericCollectionHolder<String>
        implements StraightCollectionHolderForTest<String, GenericCollectionHolder_SizeAlias> {

    public int amountOfCall = 0;

    public GenericCollectionHolder_SizeAlias() { super(ABCD); }

    @Override public int getAmountOfCall() { return amountOfCall; }

    @Contract(ALWAYS_THIS_1)
    @Override public GenericCollectionHolder_SizeAlias execute(final Consumer<GenericCollectionHolder_SizeAlias> action) {
        action.accept(this);
        return this;
    }

    @Override public @Range(from = 0, to = MAX_INT_VALUE) int size() {
        amountOfCall++;
        return super.size();
    }

}
